src/Persistent/Implementation/AcidState.hs

```diff
@@ -50,21 +51,18 @@ mkRunPersistOnDisk cfg =
   where
     opn aulaData = do
         st <- openLocalStateFrom (cfg ^. persistConfig . dbPath) aulaData
-        tid <- forkIO . forever $ do
-            let run = do
-                    logger cfg "[create acid-state checkpoint, archive]"
-                    createCheckpoint st
-                    createArchive st
+        let delay_min = cfg ^. persistConfig . snapshotIntervalMinutes
+        let delay_us = delay_min * 1000000 * 60
 
-                catchAll (SomeException e) = do
-                    logger cfg ("error creating checkpoint or archiving changelog: " <> show e)
-
-                delay_min = cfg ^. persistConfig . snapshotIntervalMinutes
-                delay_us = delay_min * 1000000 * 60
-
-            run `catch` catchAll
-            threadDelay delay_us
+        let checkpoint = do
+                logger cfg "[create acid-state checkpoint, archive]"
+                createCheckpoint st
+                createArchive st
+        let logException (SomeException e) = do
+                logger cfg ("error creating checkpoint or archiving changelog: " <> show e)
 
+        let deamon = timeoutDaemon (logger cfg) "checkpoint" delay_us checkpoint logException
+        tid <- deamon ^. timeoutDaemonStart
```

Comment:
You're not making use of the `Daemon` class here, though.  The reason for that may be that this class is not particularly useful.  I might have missed something (I have before), but you should either come up with a reason for it (at least a potential, future one) or remove it.